ml-agents/mlagents/trainers/torch/layers.py

```diff
@@ -120,14 +120,21 @@ class LinearEncoder(torch.nn.Module):
     Linear layers.
     """
 
-    def __init__(self, input_size: int, num_layers: int, hidden_size: int):
+    def __init__(
+        self,
+        input_size: int,
+        num_layers: int,
+        hidden_size: int,
+        kernel_init: Initialization = Initialization.KaimingHeNormal,
+        kernel_gain: float = 1.0,
```

Comment:
Would it make sense to default this to None and than have something like
```
if kernel_gain is None:
  if kernel_init == Initialization.KaimingHeNormal:
    kernel_gain = 1.0
  elif kernel_init == Initialization.Normal:
    kernel_gain = (0.125 / hidden_size) ** 0.5
  else:
    kernel_gain = 1.0
```
?

Reply:
Either that or have static methods to create a LinearEncoder that have different gain defaults?

Reply:
it might make sense to put that in `linear_layer`. Personally, my take is to just provide the args since anyone playing with them is probably an advanced user but I am open to this.

Thoughts @vincentpierre @ervteng

Reply:
I don't have strong feelings on it. Maybe better to wait until we have >2 data points to extrapolate from.